### PR TITLE
[path_provider] Fix links in READMEs

### PR DESCRIPTION
**Pull request by stuartmorgan**
Most of the implementation packages' READMEs had broken markdown for
what was supposed to be a link to the app-facing package.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/path_provider/path_provider_macos/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-# 2.0.2
+## 2.0.3
+
+* Fix link in README
```
Comment (bparrishMines) [inline packages/path_provider/path_provider_macos/CHANGELOG.md:3]: nit:
```suggestion
* Fix link in README.
```

File: packages/path_provider/path_provider_linux/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 2.1.2
+
+* Fix link in README
```
Comment (bparrishMines) [inline packages/path_provider/path_provider_linux/CHANGELOG.md:3]: nit:
```suggestion
* Fix link in README.
```

File: packages/path_provider/path_provider_ios/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 2.0.7
+
+* Fix link in README
```
Comment (bparrishMines) [inline packages/path_provider/path_provider_ios/CHANGELOG.md:3]: nit:
```suggestion
* Fix link in README.
```

File: packages/path_provider/path_provider_android/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 2.0.7
+
+* Fix link in README
```
Comment (bparrishMines) [inline packages/path_provider/path_provider_android/CHANGELOG.md:3]: nit:
```suggestion
* Fix link in README.
```

File: packages/path_provider/path_provider_linux/CHANGELOG.md [hunk 5]
```diff
@@ -46,5 +50,5 @@
 * This updates the readme and pubspec and example to reflect the endorsement of this implementation of `path_provider`
 
 ## 0.0.1
-* The initial implementation of path_provider for Linux
+* The initial implementation of path\_provider for Linux
```
Comment (bparrishMines) [inline packages/path_provider/path_provider_linux/CHANGELOG.md:53]: As a side question, why did we start to include the escape in `path\_provider`?

**Review by bparrishMines — APPROVED** [review]
LGTM with small nits and a question

File: packages/path_provider/path_provider_macos/CHANGELOG.md (hunk 1 above)
Comment (stuartmorgan) [reply packages/path_provider/path_provider_macos/CHANGELOG.md]: Fixed in all READMEs (and documented in a new style guide).

File: packages/path_provider/path_provider_linux/CHANGELOG.md (hunk 5 above)
Comment (stuartmorgan) [reply packages/path_provider/path_provider_linux/CHANGELOG.md:53]: `_` is special markdown syntax (for italics), and one of my editors flags unescaped uses of it as a literal character (outside of ` `` `s), so I've been fixing it as I encounter it. The rules for exactly how the parser decides if an unescaped use is literal or not aren't entirely clear to me, so it seemed like it was better to err or the side of escaping it.

**Pull request merged**